### PR TITLE
[PM-1379] add DeviceTrustCryptoService with establish trust logic

### PR DESCRIPTION
**Pull request by jlf0dev**
## Type of change
- [ ] Bug fix
- [x] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other

## Objective

Adds the Establish Trust logic to mobile. More information and clients equivalent in bitwarden/clients#5339

## Code changes

- **DeviceTrustCryptoService:** Will be responsible for handling the device specific key related methods, such as the trusted device ones in there.

## Before you submit
- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: src/Core/Services/ApiService.cs
Comment (andrebispo5) [inline src/Core/Services/ApiService.cs:595]: `HttpMethod.Get` should be `POST` or `PUT`

Comment (jlf0dev) [reply src/Core/Services/ApiService.cs]: Great catch! Updated

**Review by andrebispo5 — APPROVED** [review]
LGTM 👍

File: src/Core/Services/DeviceCryptoService.cs
Comment (fedemkr) [inline src/Core/Services/DeviceCryptoService.cs:58]: ⛏️ Remove the use of `this` (I thought this was caught by the formatting tool)

⛏️ IMO it's easier to read and understand what part is by deconstructing the tuple, so instead of `deviceKeyPair` you can have `var (devicePublicKey, devicePrivateKey)` and you can remove the comment given that it's more readable this way. And on the usage you can directly use `devicePublicKey` instead of `deviceKeyPair.Item1`.

🤔  Is there a reason to defer the encrypt task executions? Also you could run the three tasks in parallel using `Task.WhenAll(...)` and it gives back the results of all the operations. On the other hand, you can directly `await` on the spot, so instead of creating task variables you can have the awaited result there to be use afterwards in the device request.

⛏️ You can remove the parenthesis `()` from the `new TrustedDeviceKeyRequest` if you want as well.

File: src/Core/Services/DeviceCryptoService.cs
Comment (fedemkr) [inline src/Core/Services/DeviceCryptoService.cs:32]: ❓ I'm not sure where this will be used, but just wanted to ask: Does this need any special exceptions handling or is it ok to just pass it through the caller as it's now?

**Review by fedemkr — CHANGES_REQUESTED** [review]

File: src/Core/Services/DeviceCryptoService.cs
Comment (jlf0dev) [reply src/Core/Services/DeviceCryptoService.cs]: Done

File: src/Core/Services/DeviceCryptoService.cs
Comment (jlf0dev) [reply src/Core/Services/DeviceCryptoService.cs]: I was attempting to execute the tasks while we do the rest of the work, but its not worth it. Removed.

File: src/Core/Services/DeviceCryptoService.cs
Comment (jlf0dev) [reply src/Core/Services/DeviceCryptoService.cs]: Not yet, this is a helper PR to unblock other work. We'll iterate further if needed.

File: src/Core/Services/DeviceCryptoService.cs
Comment (jlf0dev) [reply src/Core/Services/DeviceCryptoService.cs]: Do we have documentation in the docs that we don't prefer `this`? If not we need to add that for consistency.

Comment (fedemkr) [reply src/Core/Services/DeviceCryptoService.cs]: I don't think we have docs for that, it'd really cool to add it. I was just guided on the general use (or not use hehe) of it in the app.

File: src/Core/Services/DeviceTrustCryptoService.cs
Comment (fedemkr) [inline src/Core/Services/DeviceTrustCryptoService.cs:42]: ⛏️ IMO this method shouldn't exist, and one should directly use the state service one, given that it's not adding/modifying any of the logic, it's private and it has the same name. However, maybe you've added it as a preparation because afterwards it will have more logic, so if that's the case it's fine.

**Review by fedemkr — APPROVED** [review]
LGTM, just detected one small thing but nothing to block the PR.

**Pull request merged**
